e2e/cypress/support/api_commands.js

```diff
@@ -1049,3 +1049,131 @@ function formRequest(method, url, formData) {
         });
     });
 }
+
+/**
+ * Creates a bot directly via API
+ * This API assume that the user is logged in and has cookie to access
+ * @param {String} username - The bots username
+ * @param {String} displayName - The non-unique UI name for the bot
+ * @param {String} description - The description of the bot
+ * All parameters are required
+ */
+Cypress.Commands.add('apiCreateBot', (username, displayName, description) => {
+    return cy.request({
+        headers: {'X-Requested-With': 'XMLHttpRequest'},
+        url: '/api/v4/bots',
+        method: 'POST',
+        body: {
+            username: username,
+            display_name: displayName,
+            description: description,
+        },
+    }).then((response) => {
+        expect(response.status).to.equal(201);
+        return cy.wrap(response.body.user_id);
+    });
+});
+
+/**
+ * Get access token
+ * This API assume that the user is logged in and has cookie to access
+ * @param {String} user_id - The user id to generate token for
+ * @param {String} description - The description of the token usage
+ * All parameters are required
+ */
+Cypress.Commands.add('apiAccessToken', (user_id, description) => {
+    return cy.request({
+        headers: {'X-Requested-With': 'XMLHttpRequest'},
+        url: '/api/v4/users/' + user_id +'/tokens',
+        method: 'POST',
+        body: {
+            description: description,
+        },
+    }).then((response) => {
+        expect(response.status).to.equal(200);
+        return cy.wrap(response.body.token);
+    });
+});
+
+/**
+ * Post a message as a bot
+ * This API assume that the user is logged in and has cookie to access
+ * @param {String} channelid - The channel to post to
+ * @param {String} message - The message of the bot
+ * @param {String} pretext - The pretext of the message
+ * @param {String} text - The text to attach to message
+ * @param {String} token - The auth token
+ * All parameters are required
+ */
+Cypress.Commands.add('apiPostBotMessage', (channelId, message, pretext, text, botUserId, token) => {
+    return cy.request({
+        headers: {'X-Requested-With': 'XMLHttpRequest'},
+        url:'api/v4/posts',
+        method: 'POST',
+        auth: {
+            bearer: token,
+          },
+        body: {
+            channel_id: {
+                        id: channelId,
+                        creator_id: botUserId,
+                      },
```

Comment:
As per documentation, this should only be channel_id, i.e. 
```channel_id: channelId,```
botUserId is not required here. I tried it with the correct params and it worked fine.

Reply:
@srkgupta, it work when this param is removed, however the message is posted a regular user instead of as a bot. Do you happen to have service to post message as bot?